metric_learn/sdml.py

```diff
@@ -285,6 +331,39 @@ class SDML_Supervised(_BaseSDML, TransformerMixin):
     The linear transformation ``L`` deduced from the learned Mahalanobis
     metric (See function `components_from_metric`.)
 
+  Methods
```

Comment:
I may have missed it in your PR description, but why do we need a full method listing in each class docstring? I expect the doc building process to handle method listings automatically.

Reply:
I expected that as well but it does not happen. So it all started because there is **a lot** of warnings like these:
```bash
/home/max/inria/metric-learn/metric_learn/sdml.py:docstring of metric_learn.sdml.SDML_Supervised:82: WARNING: autosummary: stub file not found 'metric_learn.SDML_Supervised.get_params'. Check your autosummary_generate setting.
```
See these [image](https://imgur.com/a/1ohSdHT) as reference.

And I've found the solution in [stackoverflow](https://stackoverflow.com/questions/65198998/sphinx-warning-autosummary-stub-file-not-found-for-the-methods-of-the-class-c) from a person that was also extending from the `Estimator` class and had exactly the same errors. So I followed his advice and added the `Method` directive in every class, and that removed all these bunch of warnings.

Without explicitly adding this directive, no method summary is shown in the docs, just the detailed method list.

I also thought that the `Method` directive could be incremental, so for each class I only had to add the methods description of that class, and not the method description of his parent, and grandparent class as well. But sadly it doesn't work that way either.

It may look like an overkill right now, but thinking about it, when someone has to add another algorithm in the future, you already have the docstrings from other algorithms as reference, so it won't be overwhelming.

Also, if someone wants to extend the docs, it's better for them to see the warnings/errors that only they are producing, because right now its really really hard to debug the docs with all these warnings. @wdevazelhes says it at #305 as well (in this case for the 'any' flag, the second bunch of warnings).

Reply:
Regarding this: how about the second answer in the stackoverflow post you linked: https://stackoverflow.com/questions/65198998/sphinx-warning-autosummary-stub-file-not-found-for-the-methods-of-the-class-c
This could provide a way to remove the warnings. We would not have any method summary but perhaps it is not big deal, considering the effort required to have this (of course the best solution would be that it is constructed automatically, but you seem to say that it is not possible)

Reply:
I've been hours looking to build the methods summary automatically without any warning, without success.

We either use `numpydoc_show_class_members = False` removing the warnings but loosing the automatic method summary, or

We keep the implicit `numpydoc_show_class_members = True` generating the automatic method summary but with a bunch of warnings.

Yes, I realized that the method summary its actually being generated with the docs as they are. So I will remove the hard coded method summary that I've made, as it's a lot of work and can be done automatically, despite the warnings.